pkg/provisioner/ironic/ironic.go

```diff
@@ -1682,6 +1682,16 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 	return operationContinuing(0)
 }
 
+// Detach removes the host from the provisioning system.
+// Similar to Delete, but ensures non-interruptive behavior
+// for the target system.  It may be called multiple times,
+// and should return true for its dirty  flag until the
+// deletion operation is completed.
+func (p *ironicProvisioner) Detach() (result provisioner.Result, err error) {
+	// Currently the same behavior as Delete()
+	return p.Delete()
```

Comment:
https://github.com/metal3-io/baremetal-operator/issues/410 wants us to power off the node as part of Delete(). Fix is being worked on in https://github.com/metal3-io/baremetal-operator/pull/816, and a hardPowerOff() performed during a Delete(). I don't think we want that behavior for Detach().

Reply:
The PowerOff() function now has an argument to force a hard shutdown, so we could just put the power off logic in the controller rather than the provisioner now. That's probably the right thing to do anyway, since wanting to power off is not provisioner-specific.

Failing that we can flip the implementations (so that Delete() powers off and then calls Detach()) but let's deal with that when we come to it. This looks fine for now.

Reply:
Right, since we've got this new provisioner API it enables the behavior to diverge in future, but right now the behavior is the same so I just call Delete()